packages/task/src/browser/task-templates.ts

```diff
@@ -0,0 +1,164 @@
+        '\t\t\t"problemMatcher": "$msCompile"',
+        '\t\t}',
+        '\t]',
+        '}'
+    ].join('\n')
+};
+
+const msbuild: TaskTemplateEntry = {
+    id: 'msbuild',
+    label: 'MSBuild',
+    autoDetect: false, // not supported in Theia
+    description: 'Executes the build target',
+    content: [
+        '{',
+        '\t// See https://go.microsoft.com/fwlink/?LinkId=733558',
+        '\t// for the documentation about the tasks.json format',
+        '\t"version": "2.0.0",',
+        '\t"tasks": [',
+        '\t\t{',
+        '\t\t\t"label": "build",',
+        '\t\t\t"type": "shell",',
+        '\t\t\t"command": "msbuild",',
+        '\t\t\t"args": [',
+        '\t\t\t\t// Ask msbuild to generate full paths for file names.',
+        '\t\t\t\t"/property:GenerateFullPaths=true",',
+        '\t\t\t\t"/t:build",',
+        '\t\t\t\t// Do not generate summary otherwise it leads to duplicate errors in Problems panel',
+        '\t\t\t\t"/consoleloggerparameters:NoSummary"',
+        '\t\t\t],',
+        '\t\t\t"group": "build",',
+        '\t\t\t"presentation": {',
+        '\t\t\t\t// Reveal the output only if unrecognized errors occur.',
+        '\t\t\t\t"reveal": "silent"',
+        '\t\t\t},',
+        '\t\t\t// Use the standard MS compiler pattern to detect errors, warnings and infos',
+        '\t\t\t"problemMatcher": "$msCompile"',
+        '\t\t}',
+        '\t]',
+        '}'
+    ].join('\n')
+};
+
+const maven: TaskTemplateEntry = {
+    id: 'maven',
+    label: 'maven',
+    sort: 'MVN',
+    autoDetect: false, // not supported in Theia
+    description: 'Executes common maven commands',
+    content: [
+        '{',
+        '\t// See https://go.microsoft.com/fwlink/?LinkId=733558',
+        '\t// for the documentation about the tasks.json format',
+        '\t"version": "2.0.0",',
+        '\t"tasks": [',
+        '\t\t{',
+        '\t\t\t"label": "verify",',
+        '\t\t\t"type": "shell",',
+        '\t\t\t"command": "mvn -B verify",',
+        '\t\t\t"group": "build"',
+        '\t\t},',
+        '\t\t{',
+        '\t\t\t"label": "test",',
+        '\t\t\t"type": "shell",',
+        '\t\t\t"command": "mvn -B test",',
+        '\t\t\t"group": "test"',
+        '\t\t}',
+        '\t]',
+        '}'
+    ].join('\n')
+};
+
+const command: TaskTemplateEntry = {
+    id: 'externalCommand',
+    label: 'Others',
+    autoDetect: false, // not supported in Theia
+    description: 'Example to run an arbitrary external command',
+    content: [
+        '{',
+        '\t// See https://go.microsoft.com/fwlink/?LinkId=733558',
+        '\t// for the documentation about the tasks.json format',
+        '\t"version": "2.0.0",',
+        '\t"tasks": [',
+        '\t\t{',
+        '\t\t\t"label": "echo",',
+        '\t\t\t"type": "shell",',
+        '\t\t\t"command": "echo Hello"',
+        '\t\t}',
+        '\t]',
+        '}'
+    ].join('\n')
+};
+
+export function getTemplates(): QuickPickItem<TaskTemplateEntry>[] {
```

Comment:
please follow our guidelines: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#di-function-export

Functionality has to be provided via DI to make it extensible and customizable even it is copied from somewhere.